loader/src/utils/web.cpp

```diff
@@ -406,7 +414,14 @@ WebTask WebRequest::send(std::string_view method, std::string_view url) {
                 if (value.ends_with('\r')) {
                     value = value.substr(0, value.size() - 1);
                 }
-                headers.insert_or_assign(key, value);
+                // Create a new vector and add to it or add to an already existing one
+                if (headers.contains(key)) {
+                    headers.at(key).push_back(value);
+                } else {
+                    std::vector<std::string> headerValues;
+                    headerValues.push_back(value);
+                    headers.insert_or_assign(key, headerValues);

```

Comment:
should probably move the vector instead of copying here

Reply:
you can also just construct it in place

Reply:
done